StardewArchipelago/Locations/CodeInjections/Vanilla/PhoneInjections.cs

```diff
@@ -65,5 +97,439 @@ public static bool CallAdventureGuild_AllowRecovery_Prefix(DefaultPhoneHandler _
                 return true; // run original logic
             }
         }
+
+        public static bool AnswerDialogueAction_AdventureGuildEntrance_Prefix(GameLocation __instance, string questionAndAnswer, string[] questionParams)

```

Comment:
I use a pattern everywhere in the code, for my patches, and I would like you to follow it as well please.
1: Always add a comment above a patch that shows the full signature of the patched method
2: All patches should be wrapped in a try{}catch{} block that logs the error and patch name when it fails
3: All prefixes that return true or false have a comment `// run original logic` or `// don't run original logic` as a visual reminder that this isn't the actual return type of the method, but a behavior of the patch itself, and if needed, we need to set the return value in addition to it.
4: If several patches are applied to the same method, under the same conditions, they should be merged into one patch instead, but then inner methods must be used to keep things separate anyway. This is because each new patch being applied has a significant performance cost.